DEVR-3579 Development of Forestplot with KM plot

fukuhiro2023 opened the pull request:
This PR adds new functions, `prepare_hr_forestly()`, `format_hr_forestly()`, and `hr_forestly()` to create a forest plot with KM plot. Some components are carried over from `forestly` package to create it. Also, the logic of event is corrected for `kmcurvely()`.

LittleBeannie reviewed (commented):
Thank you for getting so much work done in this PR! I only have 2 minor comments:

1. For the y-axis of the KM plots, could we rename it to something like "Survival probability" (as default) and allow users to customize the y-axis label in `format_hr_forestly()`?
2. When running the interactive HR forest plot, I noticed many messages about joining tables. Could we suppress these messages?

<img width="663" height="377" alt="image" src="https://github.com/user-attachments/assets/04087bb7-ba50-46d0-a6ac-41a577bf2b7d" />

Thanks again for your efforts!

LittleBeannie commented:
I created 3 new issues (https://github.com/MSDLLCpapers/kmcurvely/issues/16, https://github.com/MSDLLCpapers/kmcurvely/issues/17, https://github.com/MSDLLCpapers/kmcurvely/issues/18) following this PR. Let’s focus on getting this PR merged first, and then we can address the 3 new issues in separate PRs.

fukuhiro2023 commented:
Thanks @LittleBeannie for your comment.
> 1. For the y-axis of the KM plots, could we rename it to something like "Survival probability" (as default) and allow users to customize the y-axis label in `format_hr_forestly()`?

Sure, updated.

> 2. When running the interactive HR forest plot, I noticed many messages about joining tables. Could we suppress these messages?

Sure, updated. This is because `by` variable is not specified in `dplyr::left_join()`.

For issues #17 and #18, I have also updated on this branch to meet the requests. Please feel free to let me know if you have any further comments. Since updating #16 may take some time, it might be better to work on it after merging this as you mentioned.
